src/main.c

```diff
@@ -166,6 +164,37 @@ int main(void) {
 #elif defined(SAMD51)
     WDT->CTRLA.reg = 0;
     while(WDT->SYNCBUSY.reg) {}
+
+    // Enable 2.7V brownout detection. The default fuse value is 1.7
+    // Set brownout detection to ~2.7V. Default from factory is 1.7V,
+    // which is too low for proper operation of external SPI flash chips (they are 2.7-3.6V).
+    // Also without this higher level, the SAMD51 will write zeros to flash intermittently.
+    // Disable while changing level.
+    SUPC->BOD33.bit.ENABLE = 0;
```

Comment:
I think you may need to busy spin on B33SRDY after changing ENABLE. The BOD33 register is documented as write-synchronized.